--- extundelete-0.2.4/src/extundelete.cc.orig	2014-12-24 14:00:40.401233636 +0100
+++ extundelete-0.2.4/src/extundelete.cc	2014-12-25 01:29:09.076514586 +0100
@@ -1777,6 +1777,23 @@ errcode_t restore_inode(ext2_filsys fs,
 			retval = EU_RESTORE_FAIL;
 		}
 	}
+	else if (LINUX_S_ISDIR(inode->i_mode)) {
+	        if(mkdir((outputdir2 + fname2).c_str(), 0700)) {
+    		        struct timeval times[2];
+            		chmod((outputdir + fname2).c_str(),inode->i_mode);
+            		lchown((outputdir + fname2).c_str(),inode->i_uid,inode->i_gid);
+            		times[0].tv_sec=inode->i_atime; times[0].tv_usec=0;
+            		times[1].tv_sec=inode->i_mtime; times[1].tv_usec=0;
+    		        utimes((outputdir + fname2).c_str(),times);
+	                std::cout << "Restored inode " << ino << " as directory ";
+        	        std::cout << (outputdir + fname2) << " deleted " << asctime(localtime((time_t*)&inode->i_dtime));
+    	        	retval = 0;
+    		} else {
+	                std::cout << "Failed to restore inode " << ino << " as directory ";
+            		std::cout << (outputdir + fname2) << " deleted " << asctime(localtime((time_t*)&inode->i_dtime));
+            		retval = EU_RESTORE_FAIL;
+    		}
+	}
 	else {
 		Log::info << "extundelete identified inode " << ino << " as "
 		<< (outputdir + fname2) << ":"
